### PR TITLE
tuning: add tune-operating-system

### PR DESCRIPTION
**Pull request by ethercflow**
Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### What is changed, added or deleted? (Required)

Add system performance tuning doc to guide how to tuning linux system.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: tune-operating-system.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+### perf 
+
+Perf 是 Linux 内核提供的一个重要的性能分析工具，它涵盖硬件级别（CPU / PMU，性能监视单元）功能和软件功能（软件计数器，跟踪点）。 详细用法请参考：<http://www.brendangregg.com/perf.html#Background> 。
+
+### BCC/bpftrace
+
+Centos 从 7.6 版本起，内核已实现对 bpf 的支持，因此可根据上述清单的结果，选取适当的工具进行深入分析。相比 perf/ftrace，bpf 提供了可编程能力，和更小的性能开销。相比 kprobe，bpf 提供了更高的安全性。更适合在生产环境上使用。关于 BCC 工具集的使用请参考：<https://github.com/iovisor/bcc/blob/master/README.md> 。
+
+## 性能调优
```
Comment (zhouqiang-cl) [inline tune-operating-system.md:45]: 要不要加上绑核

File: tune-operating-system.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+- `dirty_ratio` 百分比值。当脏的 page cache 总量达到系统内存总量的这一百分比后，系统将开始使用 pdflush 操作将脏的 page cache 写入磁盘。默认值为20％，通常不需调整。对于高性能 SSD，比如 NVMe 设备来说，降低其值有利于提高内存回收时的效率。
+- `dirty_background_ratio` 百分比值。当脏的 page cache 总量达到系统内存总量的这一百分比后，系统开始在后台将脏的 page cache 写入磁盘。默认值为10％，通常不需调整。对于高性能 SSD，比如 NVMe 设备来说，设置较低的值有利于提高内存回收时的效率。
+
+### 存储及文件系统
+
+内核 I/O 栈链路较长，包含了文件系统层、块设备层和驱动层。
+
+#### I/O 调度器
+
+I/O 调度程序确定 I/O 操作何时在存储设备上运行以及持续多长时间。也称为 I/O 升降机。对于 SSD 设备，宜设置为 noop。
```
Comment (zhouqiang-cl) [inline tune-operating-system.md:80]: 默认是什么值

File: tune-operating-system.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+#### 格式化参数 —— 块大小
+
+块是文件系统的工作单元。块大小决定了单个块中可以存储多少数据，因此决定了一次写入或读取的最小数据量。
+默认块大小适用于大多数使用情况。但是，如果块大小（或多个块的大小）与通常一次读取或写入的数据量相同或稍大，则文件系统将性能更好，数据存储效率更高。小文件仍将使用整个块。文件可以分布在多个块中，但这会增加运行时开销。
+使用 mkfs 命令格式化设备时，将块大小指定为文件系统选项的一部分。指定块大小的参数随文件系统的不同而不同。有关详细信息，请查询对应文件系统的 mkfs 手册页，比如 `man mkfs.ext4`。
+
+#### 挂载参数
+
+`noatime` 读取文件时，将禁用对元数据的更新。它还启用了nodiratime 行为，该行为会在读取目录时禁用对元数据的更新。
```
Comment (zhouqiang-cl) [inline tune-operating-system.md:94]: noatime 官方有性能数据吗

File: tune-operating-system.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+aliases: ['/docs-cn/dev/reference/performance/tune-system/']
+---
+
+# 操作系统性能参数调优
+
+本文档仅用于描述如何优化 Centos 7 的各个子系统。
+
+> **注意：**
+> 
+> 1. Centos 7 操作系统的默认配置适用于中等负载下运行的大多数服务。调整特定子系统的性能可能会对其他子系统产生负面影响。因此在调整系统之前，请备份所有数据和配置信息；
```
[the remark this answers is not in the document]
Comment (ethercflow) [reply tune-operating-system.md:13]: 嗯，所有数据指的是用户数据。

File: tune-operating-system.md (hunk 3 above)
Comment (ethercflow) [reply tune-operating-system.md:94]: 没有，具体开销主要取决于盘的性能。

File: tune-operating-system.md (hunk 2 above)
Comment (ethercflow) [reply tune-operating-system.md:80]: 默认值和盘有关，比如咱们 IDC 的 sda 会设置成 deadline，nvme 默认是 noop。

File: tune-operating-system.md (hunk 1 above)
Comment (ethercflow) [reply tune-operating-system.md:45]: @zhouqiang-cl done, PTAL.

File: tune-operating-system.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (yikeke) [reply tune-operating-system.md:13]:
```suggestion
> 1. Centos 7 操作系统的默认配置适用于中等负载下运行的大多数服务。调整特定子系统的性能可能会对其他子系统产生负面影响。因此在调整系统之前，请备份所有用户数据和配置信息；
```
那么改一下？

File: tune-operating-system.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+---
+
+# 操作系统性能参数调优
+
+本文档仅用于描述如何优化 Centos 7 的各个子系统。
+
+> **注意：**
+>
+> 1. Centos 7 操作系统的默认配置适用于中等负载下运行的大多数服务。调整特定子系统的性能可能会对其他子系统产生负面影响。因此在调整系统之前，请备份所有数据和配置信息；
+> 2. 请在测试环境下对所有修改做好充备测试后，再应用到生产环境中。
```
Comment (yikeke) [inline tune-operating-system.md:14]:
```suggestion
> 2. 请在测试环境下对所有修改做好充分测试后，再应用到生产环境中。
```

File: tune-operating-system.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+## 性能分析工具
+
+系统调优需要根据系统性能分析的结果做指导，因此本文先列出常用的性能分析方法。
+
+### 60s 分析法
+
+此分析法由《性能之巅》的作者 Brendan Gregg 及其所在的 Netflix 性能工程团队公布，所用到到工具均可从发行版的官方源获取，通过分析以下清单的输出，可定位大部分常见的性能问题。
```
Comment (yikeke) [inline tune-operating-system.md:22]:
```suggestion
此分析法由《性能之巅》的作者 Brendan Gregg 及其所在的 Netflix 性能工程团队公布，所用到的工具均可从发行版的官方源获取，通过分析以下清单中的输出，可定位大部分常见的性能问题。
```

File: tune-operating-system.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
+---
+title: 操作系统性能参数调优
+category: reference
+aliases: ['/docs-cn/dev/reference/performance/tune-system/']
```
Comment (yikeke) [inline tune-operating-system.md:4]:
```suggestion
```
A new file does not need to include an alias. The aliases are for redirecting the provided alias link.

File: tune-operating-system.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+具体用法可查询相应 man 手册。
+
+### perf
+
+Perf 是 Linux 内核提供的一个重要的性能分析工具，它涵盖硬件级别（CPU/PMU，性能监视单元）功能和软件功能（软件计数器，跟踪点）。详细用法请参考：<http://www.brendangregg.com/perf.html#Background> 。
+
+### BCC/bpftrace
+
+Centos 从 7.6 版本起，内核已实现对 bpf 的支持，因此可根据上述清单的结果，选取适当的工具进行深入分析。相比 perf/ftrace，bpf 提供了可编程能力，和更小的性能开销。相比 kprobe，bpf 提供了更高的安全性，更适合在生产环境上使用。关于 BCC 工具集的使用请参考：<https://github.com/iovisor/bcc/blob/master/README.md> 。
```
Comment (yikeke) [inline tune-operating-system.md:43]:
```suggestion
Centos 从 7.6 版本起，内核已实现对 bpf 的支持，因此可根据上述清单的结果，选取适当的工具进行深入分析。相比 perf/ftrace，bpf 提供了可编程能力，和更小的性能开销。相比 kprobe，bpf 提供了更高的安全性，更适合在生产环境上使用。关于 BCC 工具集的使用请参考 [BPF Compiler Collection (BCC)](https://github.com/iovisor/bcc/blob/master/README.md)。
```

File: tune-operating-system.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+7. free -m
+8. sar -n DEV 1
+9. sar -n TCP,ETCP 1
+10. top
+
+具体用法可查询相应 man 手册。
+
+### perf
+
+Perf 是 Linux 内核提供的一个重要的性能分析工具，它涵盖硬件级别（CPU/PMU，性能监视单元）功能和软件功能（软件计数器，跟踪点）。详细用法请参考：<http://www.brendangregg.com/perf.html#Background> 。
```
Comment (yikeke) [inline tune-operating-system.md:39]:
```suggestion
Perf 是 Linux 内核提供的一个重要的性能分析工具，它涵盖硬件级别（CPU/PMU，性能监视单元）功能和软件功能（软件计数器，跟踪点）。详细用法请参考 [perf Examples](http://www.brendangregg.com/perf.html#Background)。
```

File: tune-operating-system.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+### 处理器——动态节能技术
+
+cpufreq 是一个动态调整 CPU 频率的模块，可支持五种模式。为保证服务性能应选用 performance 模式，将 CPU 频率固定工作在其支持的最高运行频率上，不进行动态调节。命令为 `cpupower frequency-set --governor performance`。
+
+### 处理器——中断亲和性
+
+- 自动平衡：可通过 `irqbalance` 服务实现。
+- 手动平衡：
+    1. 确定需要平衡中断的设备，从 Centos 7.5 开始，系统会自动为某些设备及其驱动程序配置最佳的中断关联性。不能再手动配置其亲和性。目前已知的有使用 `be2iscsi` 驱动的设备，以及 NVME 设置；
+    2. 对于其他设备，可查询其芯片手册，是否支持分发中断，若不支持，则该设备的所有中断会路由到同一个 CPU 上，无法对其进行修改，若支持，则计算 `smp_affinity` 掩码并设置对应的配置文件，具体请参考内核文档：<https://www.kernel.org/doc/Documentation/IRQ-affinity.txt>。
```
Comment (yikeke) [inline tune-operating-system.md:58]:
```suggestion
    2. 对于其他设备，可查询其芯片手册，是否支持分发中断，若不支持，则该设备的所有中断会路由到同一个 CPU 上，无法对其进行修改，若支持，则计算 `smp_affinity` 掩码并设置对应的配置文件，具体请参考[内核文档](https://www.kernel.org/doc/Documentation/IRQ-affinity.txt)。
```

File: tune-operating-system.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+### 网络
+
+网络子系统由具有敏感连接的许多不同部分组成。因此，Centos 7 网络子系统旨在为大多数工作负载提供最佳性能，并自动优化其性能。因此，通常无需手动调整网络性能。
+
+网络问题通常由硬件或相关设施出现问题导致的，因此在调优协议栈前，请先排除硬件问题。
+
+尽管网络堆栈在很大程度上是自我优化的，但是在网络数据包处理过程中有许多要点可能会成为瓶颈并降低性能：
+
+- 网卡硬件缓存：正确观察硬件层面的丢包方法是使用 `ethtool -S ${NIC_DEV_NAME}` 命令观察 drops 字段。当出现丢包现象时，主要考虑是硬/软中断的处理速度跟不上网卡接收速度。若接收缓存小于最大限制时，也可尝试增加 RX 缓存来防止丢包。查询命令为：`ethtool -g ${NIC_DEV_NAME}`，修改命令为 `ethtool -G ${NIC_DEV_NAME}`。
+- 硬中断：若网卡支持 Receive-Side Scaling（RSS 也称为多网卡接收）功能，则观察 /proc/interrrputs 网卡中断，如果出现了中断不均衡的情况，请参考处理器调优章节。若不支持 RSS 或 RSS 数量远小于物理 CPU 核数，则可配置 Receive Packet Steering（RPS，可以看作 RSS 的软件实现），及 RPS 的扩展 Receive Flow Steering (RFS)。具体设置请参考内核文档：<https://www.kernel.org/doc/Documentation/networking/scaling.txt> 。
```
Comment (yikeke) [inline tune-operating-system.md:111]:
```suggestion
- 硬中断：若网卡支持 Receive-Side Scaling（RSS 也称为多网卡接收）功能，则观察 /proc/interrrputs 网卡中断，如果出现了中断不均衡的情况，请参考处理器调优章节。若不支持 RSS 或 RSS 数量远小于物理 CPU 核数，则可配置 Receive Packet Steering（RPS，可以看作 RSS 的软件实现），及 RPS 的扩展 Receive Flow Steering (RFS)。具体设置请参考[内核文档](https://www.kernel.org/doc/Documentation/networking/scaling.txt)。
```

**Pull request merged**